components/server/src/user/token-service.ts

```diff
@@ -29,6 +31,17 @@ export class TokenService implements TokenProvider {
 
     async getTokenForHost(user: User | string, host: string): Promise<Token | undefined> {
         const userId = User.is(user) ? user.id : user;
+
+        // EXPERIMENT(sync_refresh_token_exchange)
+        const syncRefreshTokenExchange = await getExperimentsClientForBackend().getValueAsync(
+            "sync_refresh_token_exchange",
```

Comment:
retry_refresh_token_exchange and sync_refresh_token_exchange are different

- sync_refresh_token_exchange will store them in database
- retry_refresh_token_exchange will revoke (internally) token after it's used

Is it intent?

Reply:
The intention behind both paths is the same. Deep down they both call:
```
await authProvider.refreshToken(user); // updating/inserting a fresh token into the DB as side effect
return await this.userDB.findTokenForIdentity(identity);   // return said token
```

It's just that the existing path has this poor-man's sync approach (from before we had redis), which I did not want for the pure redis solution.
What I basically did for `sync_refresh_token_exchange` is to wrap it into a redis mutex, nothing more. The rest is noise/slight code re-arrangements.